### ISSUE TITLE
`aur build -c --pkgver` does not sync dependencies

### ISSUE THREAD

**Issue by FichteFoll** [AladW/aurutils#771]
I used to be able to build packages with `aur build -c --pkgver` on 2.x, but with 3.0 this command stopped syncing dependencies in the chroot. Simply not specifying `--pkgver` behaves as expected (although I need `-f` or have to update the PKGBUILD's `pkgrel` to make it not skip the existing build).

Apparently, this behavior was changed in 9a6a84dc8f60a9e26959b73d563e2a8457d66ce9.

**Comment by AladW** [comment]
I suspect in 2.x `aur build -c --pkgver` would install the `makedepends` both on the host and in the container?

**Comment by FichteFoll** [comment]
No, because then I would have `boost` installed on my system, which I don't. I suspect it just ignored makedepends for the host entirely, but since the vast majority of `pkgver` implementation only calls `git`, `sed` and `tr`, that will hardly come up.

**Comment by AladW** [comment]
May be related to not using `--noprepare` then: ad7055b7a19a03a92c027407b0301ee143e7f116

**Comment by foundObjects** [comment]
I'm experiencing this as well with the `cpu-x` package from the AUR. I tried to kick off a chroot build of that today after fixing its depends to include `libprocps` but it fails to build with a missing `mawk` dependency, which build should probably be able to find since it's available on the AUR.

Anyway, if you need a repro for this one `cpu-x` should reliably fail.

Here's the alias I use to kick off chroot builds:

```
arglebargle@archx16 cpu-x (fix ✓)]$ type aurbuild
aurbuild is aliased to `aur build -d "$AUR_REPO" --syncdeps --sign --chroot --directory="$AUR_BUILDROOT" --makepkg-conf=/etc/aurutils/makepkg-custom.conf'
```

I'd just use `aur sync` but I can't here since I need to patch the PKGBUILD prior to building.

**Comment by AladW** [comment]
> build should probably be able to find since it's available on the AUR.

`build` doesn't do AUR dependency resolution, you need `sync` or `depends` for this. Anyway `cpu-x` does not have a `pkgver()` function so the topic's `--pkgver` is not needed.

> I'd just use aur sync but I can't here since I need to patch the PKGBUILD prior to building.

There are several flags for patch application, see `Fetch options` in `man aur-sync`. Alternatively run `aurbuild` on the `mawk` PKGBUILD first.
